### PR TITLE
add support of width, height and variant for textarea

### PR DESCRIPTION
**Pull request by MonPote**
**Component**:

**Description**:

It add the support for width (and height) vs cols, rows.
It also add variant to handle not “code” style.

**Design**:

**Breaking Changes**:

- [] Breaking Changes

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/textarea/TextArea.component.tsx [hunk 1]
```diff
@@ -2,22 +2,42 @@ import React, { forwardRef, HTMLProps } from 'react';
...
+
+type TextAreaVariant = 'code' | 'text';
+type Props =
+  | Omit<HTMLProps<HTMLTextAreaElement>, 'as'> & { variant?: TextAreaVariant };
 type RefType = HTMLTextAreaElement | null;
 
-const TextAreaContainer = styled.textarea`
+const TextAreaContainer = styled.textarea<{
+  variant: TextAreaVariant;
+  width?: string | number;
```
Comment (ChengYanJin) [inline src/lib/components/textarea/TextArea.component.tsx]: Just thinking maybe we can use `CSSProperties` which has already typed for us ? 
```suggestion
  width?: CSSProperties['width'];
```

File: src/lib/components/textarea/TextArea.component.tsx [hunk 2]
```diff
@@ -2,22 +2,42 @@ import React, { forwardRef, HTMLProps } from 'react';
...
+  height?: string | number;
```
Comment (ChengYanJin) [inline src/lib/components/textarea/TextArea.component.tsx]:
```suggestion
  height?: CSSProperties['height'];
```

File: src/lib/components/textarea/TextArea.component.tsx [hunk 3]
```diff
@@ -2,22 +2,42 @@ import React, { forwardRef, HTMLProps } from 'react';
 import styled, { css } from 'styled-components';
 import { spacing } from '../../spacing';
 import { getTheme } from '../../utils';
-type Props = Omit<HTMLProps<HTMLTextAreaElement>, 'as'>;
+
+type TextAreaVariant = 'code' | 'text';
+type Props =
+  | Omit<HTMLProps<HTMLTextAreaElement>, 'as'> & { variant?: TextAreaVariant };
```
Comment (JBWatenbergScality) [inline src/lib/components/textarea/TextArea.component.tsx]: Does this type include the width/height props already ?  If yes does their types complies with `CSSProperties['width'];` ?

Comment (MonPote) [reply src/lib/components/textarea/TextArea.component.tsx]: Yep it does and it comply with `CSSProperties['width']`.

File: src/lib/components/textarea/TextArea.component.tsx (hunk 1 above)
Comment (MonPote) [reply src/lib/components/textarea/TextArea.component.tsx]: Indeed thanks for pointing this, much better !

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**